src/crowdin_client.h

```diff
@@ -80,49 +82,62 @@ class CrowdinClient
     struct ProjectListing
     {
         std::wstring name;
-        std::string identifier;
-        bool downloadable;
+        int id;
     };
 
     /// Retrieve listing of projects accessible to the user
     dispatch::future<std::vector<ProjectListing>> GetUserProjects();
 
+    // File information
+    struct FileInfo
+    {
+        std::wstring pathName;
+        int id, dirId, branchId;
+        
+    };
+
+
     /// Project detailed information
     struct ProjectInfo
     {
         std::wstring name;
-        std::string identifier;
+        int id;
         std::vector<Language> languages;
-        std::vector<std::wstring> files;
+        std::vector<FileInfo> files;
     };
 
     /// Retrieve listing of projects accessible to the user
-    dispatch::future<ProjectInfo> GetProjectInfo(const std::string& project_id);
+    dispatch::future<ProjectInfo> GetProjectInfo(const int project_id);
 
     /// Asynchronously download specific Crowdin file into @a output_file.
-    dispatch::future<void> DownloadFile(const std::string& project_id,
-                                        const std::wstring& file,
-                                        const Language& lang,
+    dispatch::future<void> DownloadFile(const long project_id,
+                                        const long file_id,
+                                        const wxString& file_name,
```

Comment:
Force-pushed to it original commit 0a2ceabfed0c873e0e16c6d0ec7a93705a1d43d0 changed `wxString`=>`std::wstring` in above func arg but still using `wxString` inside above funct itself. Because `wxString` provides `EndsWith()` required in above func implementation while nothing similar with `std::string` out of the box, the only below one is simplest and most straightforward one I was able to find
```
inline bool ends_with(std::string const & value, std::string const & ending)
{
    if (ending.size() > value.size()) return false;
    return std::equal(ending.rbegin(), ending.rend(), value.rbegin());
}
```
though not sure whether it worths adding above func to [str_helpers.h ](https://github.com/tetatetit/poedit/blob/crowdin-new-oauth-api-xliff/src/str_helpers.h) or otherwise please let me know is still worths...anyway if we want to completely avoid _any_ `wx` stuff in "Non-GUI" code - we  also need to avoid it here https://github.com/tetatetit/poedit/blob/79d5f05e4ac94169c3062cbe2ca98a4220ef70a7/src/crowdin_client.cpp#L398 (when i comes to replacing it with `wxURI` [as you requested in another comment](https://github.com/vslavik/poedit/pull/629#discussion_r415104340)) and I guess `wxLogTrace` as well :).

Reply:
See `boost::ends_with()` used e.g. in crowdin_gui.cpp. 

>  we also need to avoid it here https://github.com/tetatetit/poedit/blob/79d5f05e4ac94169c3062cbe2ca98a4220ef70a7/src/crowdin_client.cpp#L398 (when i comes to replacing it with wxURI as you requested in another comment) and I guess wxLogTrace as well :).

I see removal of wx code in non-GUI code as aspirational — should be done when reasonably possible, when an alternative isn't too burdensome or exists at all, and should be done in API.

As for the linked code, any such parsing by counting separator characters is fragile and a red flag (hence my wxURI suggestion in the first place).